### PR TITLE
Disable coroutine support for GCC 10.2 and earlier

### PR DESCRIPTION
**Pull request by ccotter**
GCC 10.2 coroutine support is not great. In particular, this version of GCC does not handle promise type constructors that accept the coroutine function parameters correctly. GCC passes a reference to the pre-copy object to the promise type constructor rather than a reference to the coroutine function argument after it's been copied to the coroutine frame. connect_awaitable relies on this working correctly, so we will require GCC 10.3 or newer to use coroutines with unifex.

Replaces #443

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,7 +40,13 @@ with the following compilers:
 
 This library also supports C++20 coroutines. You will need to compile with
 coroutine support enabled if you want to use the coroutine integrations.
-This generally means adding `-std=c++2a` or `-fcoroutines-ts` on Clang (see "Configuring" below).
+This generally means adding `-std=c++2a`, `-fcoroutines-ts` on Clang, or
+`-fcoroutines` for GCC (see "Configuring" below). Compilers with stable
+coroutine support are required:
+
+* GCC, 10.3 and later
+* Clang, 10.x and later
```
Comment (ccotter) [inline README.md:48]: I'm also not sure what version of clang would be sufficient to be considered "stable" to use coroutines with libunifex, so I've left it at the minimum version of support for libunifex as stated above.

File: README.md [hunk 2]
```diff
@@ -40,7 +40,13 @@ with the following compilers:
...
+* MSVC not currently supported
```
[the remark this answers is not in the document]
Comment (janondrusek) [reply README.md:49]: @lewissbaker may be able to answer the MSVC question.

I am OK with Clang, GCC targeting. Should we aim for versions supported by the compiler maintainers or current major and one previous major version?

As of today GCC lists `10.4` and Clang last released `10.x` branch is `10.0.1` from Aug 2020.

[the remark this answers is not in the document]
Comment (janondrusek) [reply README.md:49]: We dropped older GCC support in #576, should be OK to close now.

**Comment by ccotter** [comment]
Closing per https://github.com/facebookexperimental/libunifex/pull/576

**Pull request closed without merging**